tests/pipenv/Pipfile.lock

```diff
@@ -0,0 +1,281 @@
+{
```

Comment:
Is there a reason to commit lock here? I think we could leave it out

Reply:
So, that changes the previous behavior where lock file was not needed? What if we just start a new project with intention of managing the dependencies with `Pipfile`? Is it going to fail too on the first run?

Reply:
These tests are for pipenv specifically and I'm not familiar with a scenario where lock isn't expected. To my knowledge, it should always be created at first usage.

Reply:
I was looking into it a bit more, and it seems like there's not good workflow at the moment for fresh projects that do not have `Pipfile.lock` generated yet. I think one option would be to first check for existence of `Pipfile.lock`, if it's avaialble, we should run the commands as proposed in this PR. If it's not present, we should skip adding the `--keep-outdated` flag as it crashes the `lock` command when there's not `Pipfile.lock`. What do you think?

Reply:
It seems to me both commands actually require a `Pipfile.lock`

```bash
[pipenv-test]$ pipenv requirements --hash

...

FileNotFoundError: [Errno 2] No such file or directory: '/tmp/pipenv-test/Pipfile.lock'
```

At this point, I personally think a lockfile should be a prerequisite for `pipenv` support in this project, since it generally seems to be for `pipenv` itself.

Reply:
I agree that `pipenv requirements --hash` requires `Pipfile.lock`, as its generating `requirements.txt` from the lockfile. 

The problem that I'm bringing up is the fact that if we accept the PR as-is, then we're breaking the functionality for users that are using the file on a new project or want to remove `Pipfile.lock` in order to regenerate it during deploy. In previous (current) implementation, if lockfile is not present, it is generated automatically for you. In proposed implementation, you run into an error and in order to resolve it, you need to somehow manually generate `Pipfile.lock` e.g. by running `pipfile lock` command. It seems to me like this is a way worse experience and I don't see why we wouldn't want to support generating lockfile if it doesn't exist. 

What do you think @rwestergren @andidev?